entity-framework/core/modeling/bulk-configuration.md

```diff
@@ -0,0 +1,65 @@
+---
+title: Model Bulk Configuration - EF Core
+description: How to apply bulk configuration during model building in Entity Framework Core
+author: AndriySvyryd
+ms.date: 11/05/2021
+uid: core/modeling/bulk-configuration
+---
+# Model bulk configuration
+
+When an aspect needs to be configured in the same way across multiple entity types the following techniques allow to reduce code duplication and consolidate the logic.
+
+See the [full sample project](https://github.com/dotnet/EntityFramework.Docs/tree/main/samples/core/Modeling/BulkConfiguration) containing the code snippets presented below.
+
+## Metadata API
+
+Every builder object returned from <xref:Microsoft.EntityFrameworkCore.ModelBuilder> exposes a <xref:Microsoft.EntityFrameworkCore.ModelBuilder.Model> or `Metadata` property that provides a low-level access to the objects that comprise the model. In particular, there are methods that allow you to iterate over specific objects in the model and apply common configuration to them.
+
+In the following example the model contains a custom value type `Money`:
+
+[!code-csharp[Main](../../../samples/core/Modeling/BulkConfiguration/Money.cs?name=Money)]
+
+Properties of this type are not discovered by default as the current EF provider doesn't know how to map it to a database type. This snippet of `OnModelCreating` adds all properties of the type `Money` and configures a value converter to a supported type - `decimal`:
+
+[!code-csharp[Main](../../../samples/core/Modeling/BulkConfiguration/MetadataAPIContext.cs?name=MetadataAPI)]
+
+### Drawbacks of the Metadata API
+
+- Unlike Fluent API, every modification to the model needs to be done explicitly. For example, if some of the `Money` properties were configured as navigations by a convention then you need to first remove the navigation referencing the CLR property before adding an entity type property for it. [#9117](https://github.com/dotnet/efcore/issues/9117) will improve this.
+- The conventions run after each change. If you remove a navigation discovered by a convention then the convention will run again and add it back. To prevent this from happening you would need to either delay the conventions until after the property is added by calling <xref:Microsoft.EntityFrameworkCore.Metadata.Conventions.IConventionContext.DelayConventions> or to mark the CLR property as ignored using <xref:Microsoft.EntityFrameworkCore.Metadata.IMutableModel.AddIgnored%2A>.
```

Comment:
I doubt many would understand what this means :sweat_smile:

Reply:
The problem, the solution or both?

Reply:
I think both. To be clear, this isn't about your text or anything - it's just that understanding the reactivity of conventions and what that could mean is a very advanced, internal thing.

Maybe it's better to go with something vague, just to convey the "unreliability" of this mechanism (e.g. "Some changes done via the Fluent API may not result in the desired results. For example, if you remove a navigation discovered by a convention, then that convention will run again and add it back."). Otherwise we could go with this too - I just doubt it would help many people.

Reply:
We'll expand and move this in https://github.com/dotnet/EntityFramework.Docs/issues/1937